### PR TITLE
Update authoring guidelines 

### PR DESCRIPTION
**Pull request by ashygee**
This PR updates the authoring guidelines to include new documentation in response to the update of Figma's component properties feature.

### CODE REVIEW

File: docs/authoring-components.md [hunk 1]
```diff
@@ -2,51 +2,68 @@
...
 This document covers the guidelines on how to author new components, variants, and their properties within Figma. For questions and additional support, please reach out in the #primer channel in Slack.
 
+> **Edit 2022 May 13:** With the introduction of new component property features in Figma, the design systems team will be restructuring and renaming component/variant properties to closer align to the component APIs seen in code.
+
 ---
 
 ## General guidelines
 
-- Write component names in human speak. For example use `Button` instead of `btn`.
+- Write component names as reflected in code as a component name but not after utility names. For example use `Button` instead of `btn` .
```
Comment (vdepizzol) [inline docs/authoring-components.md:11]: With this updated copy it sounds like `btn` refers to an utility name, which is not the case. `btn` is just an old way we reference the component name in Primer CSS.

Maybe this bullet point could also point to the [Spelling of component names ADR](https://github.com/github/primer/blob/main/adrs/2022-02-15-spelling-of-component-names.md).

File: docs/authoring-components.md [hunk 2]
```diff
@@ -2,51 +2,68 @@
...
 
+### Component properties
+
+Certain visual options can be controlled with Figma's component properties. Apply component properties when controlling the visibility of a layer, swapping between specific instance options, or controlling the content of a text layer. Below is a list of common properties and the types of items they control. When ordering component properties, order them from the start to end to how they would be seen within the DOM. 
+
+As a reminder, variant properties will always appear above component properties in the properties panel.
+
+| Name | Description | Component property type |
+| :--: | :--: | :--: |
+| ⬅️ visual? | Show/hide the leading visual which is a the start of the component (left-side) | [Boolean](https://help.figma.com/hc/en-us/articles/5579474826519-Create-and-use-component-properties#h_01G2Q5GA6DEB604H2E5H5C5TA4) |
```
Comment (vdepizzol) [inline docs/authoring-components.md:34]: In this example there's both `⬅️ visual` for `leadingVisual`, and `leadingAction` in full. Should we normalize this with the same props used in code? Such as: `leadingVisual` and `leadingAction`.

@ashygee after we chatted about this, I tested using both "⬅️" and `<-` (which is nicely rendered by Figma as "←"), and I found that no prefix gives the maximum legibility in such a tight space:

| Prefix | Result |
|--------|--------|
| ⬅️ | Rendering |
| `<-` | Rendering |
| (None) | Rendering |

File: docs/authoring-components.md (hunk 1 above)
Comment (ashygee) [reply docs/authoring-components.md:11]: Thanks for the clarification. I will update the example and cross-reference the ADR.

File: docs/authoring-components.md (hunk 2 above)
Comment (ashygee) [reply docs/authoring-components.md:34]: We can remove them and add to the layer name only as suggested by @langermank in her comment in https://github.com/github/primer/issues/979#issuecomment-1128370017.

**Comment by ashygee** [comment]
Thank you for the feedback @vdepizzol. I've addressed your comments and have re-requested a review.

File: docs/authoring-components.md [hunk 3]
```diff
@@ -2,51 +2,110 @@
...
-- When writing variants, keep the property name in sentence case and the values in lowercase. This helps to quickly discern properties and values from each other when creating complex components containing multiple properties.
-- Using emoji arrows to show on which side an item will be placed is also helpful.
-- To enable a toggle switch, use boolean value pairs such as `on`/`off`,`true`/`false`, and `yes`/`no`. Use the value name that makes sense to the action being executed when interacting with the toggle switch. Many of these are interchangeable but there are instances when one pairing may provide better understanding to what the intention of the property is used for.
+- When properties, either variant or component, write them as they are seen or would be written in code. For example, the prop for an icon or avatar in the action list component is written as `leadingVisual`.
+- When writing variants, keep the property name in as written in code with lower or camelCase and the values in lowercase.
+- When indicating a boolean property add a "?". For example, the property to show/hide the title of a component is written as "title?"
```
[the remark this answers is not in the document]
Comment (langermank) [reply docs/authoring-components.md:85]: Oh, is this a new thing? So if you don't pass in a string its also `false`? Cool 😮 then I think the `?` does make sense? 😆 

Okay if that's the case, maybe...

```suggestion
- When indicating a string that is also a boolean property add a "?". For example, the property to show/hide the title of a component is written as "title?"
```

**Review by langermank — COMMENTED** [review]
Looks good, just a few grammar suggestions and questions

File: docs/authoring-components.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashygee) [reply docs/authoring-components.md:85]: @langermank this isn't actually a string but a boolean property set automatically from Figma. The indication of the `?` was mostly to notate that this is a toggleable selection, as in you can turn it on or off but not necessarily toggle between two different instance options.  I think the above re-write makes it a little confusing so I'm not going to commit it in this batch but let's discuss more async.

File: docs/authoring-components.md [hunk 4]
```diff
@@ -2,51 +2,110 @@
...
 | Category | Description | Included values |
 |--------|--------|--------|
-| `Type` | A variation of the component or a specific part | text, single select, multi-select, danger, header, item, footer | 
-| `State` | The state of the component's interactivity | default (rest), hover, selected, disabled | 
-| `Size` | Size variations of the component | small, medium, large OR pixel values | 
-| `⬅️ Leading  [accessory]` | The leading visual accessory of a component placed to the **left** of a text label or description | icon, avatar |
-| `➡️ Trailing [accessory]` | The trailing visual accessory of a component placed to the **right** of a text label or description | icon, avatar, label, counter |
-| `Description` | Descriptive text within a component; can be a toggle (boolean) set of values or a dropdown selection | on/off [boolean], short/long, inline/block |
-|  `Selected` | Mark the component as selected; toggle (boolean) the selection marker, typically a check mark | on/off |
+| `type` | A variation of the component or a specific part | text, single select, multi-select, danger, header, item, footer |
```
[the remark this answers is not in the document]
Comment (ashygee) [reply docs/authoring-components.md:102]: This is one of those tricky words. For a while there was a long-running [conversation](https://github.com/github/primer/issues/253#issuecomment-981881508) during the API consistency work where using the term "variant" was confusing as it meant something different in Rails for ViewComponents. Happy to update this section in the future but we should probably codify how we want to refer to these. FWIW I do also like `variant` here.

**Review by langermank — APPROVED** [review]
🚀

File: docs/authoring-components.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (langermank) [reply docs/authoring-components.md:85]: Thanks for that info! I don't see this comment as a blocker, feel free to disregard 👍

**Pull request merged**